src/io/flutter/server/vmService/VmServiceWrapper.java

```diff
@@ -249,10 +277,11 @@ public void received(final Isolate isolate) {
     }
   }
 
-  private void setInitialBreakpoints(@NotNull IsolateRef isolateRef) {
+  private void setInitialBreakpointsAndCheckExtensions(@NotNull IsolateRef isolateRef, @NotNull Isolate isolate) {
     doSetBreakpointsForIsolate(myBreakpointHandler.getXBreakpoints(), isolateRef.getId(), () -> {
       myIsolatesInfo.setBreakpointsSet(isolateRef);
     });
+    PerfService.updateRegisteredExtensions(isolate);
```

Comment:
you should be able to just get just the perf service associated with this app instead of making this static.

Reply:
Right, this is part of that bit I need to rethink.